docs/source/spark.rst

```diff
@@ -0,0 +1,164 @@
+extra serialization layers (see py4j_) and computation layers.  Limitations
+like the Java heap size and large Java stack traces come as a surprise to users
+accustomed to native code execution.
+
+Dask has an advantage for Python users because it is itself a Python library,
+so serialization and debugging when things go wrong happens more smoothly.
+
+However, dask has no benefit to non-Python users while Spark is useful in a
+variety of JVM languages (Scala, Java, Clojure) as well as limited support in
+Python and R.  New Spark projects like the DataFrame skip serialization and
+boxed execution issues by forgoing the Python process entirely and instead have
+Python code drive native Scala code.  APIs for these libraries tend to lag a
+bit behind their Scala counterparts.
+
+
+Scope
+~~~~~
+
+Spark was originally built around the RDD, an unordered collection allowing
+repeats.  Most spark add-ons were built on top of this construct, inheriting
+both its abilities and limitations.
+
+Dask is built on a lower-level and more general construct of a generic task
+graph with data dependencies.  This allows more general computations to be
+built by users within the dask framework.  This is probably the largest
+fundamental difference between the two projects.  Dask gives up high-level
+understanding to allow users to express more complex parallel algorithms.  This
+ended up being essential when writing complex projects like ``dask.array``.
+
+
+Developer-Facing Differences
+----------------------------
+
+Graph Granularity
+~~~~~~~~~~~~~~~~~
+
+Both Spark and Dask represent computations with directed acyclic graphs.  These
+graphs however represent computations at very different granularities.
+
+One operation on a Spark RDD might add a node like ``Map`` and ``Filter`` to
+the graph.  These are high-level operations that convey meaning and will
+eventually be turned into many little tasks to execute on individual workers.
+This many-little-tasks state is only available internally to the Spark
+scheduler.
+
+Dask graphs skip this high-level representation and go directly to the
+many-little-tasks stage.  As such one ``map`` operation on a dask collection
+will immediately generate and add possibly thousands of tiny tasks to the dask
+graph.
+
+This difference in the scale of the underlying graph has implications on the
+kinds of analysis and optimizations one can do and also on the generality that
+one exposes to users.  Dask is unable to perform some optimizations that Spark
+can because Dask does not have a top-down picture of the computation it was
+asked to perform.  However, dask is able to easily represent far more complex
+algorithms and expose the creation of these algorithms to normal users.
+
+Dask.bag, the equivalent of the Spark.RDD, is just one abstraction built on top
+of dask.  Others exist.  Alternatively power-users can forego high-level
+collections entirely and jump striaght to direct, low-level, task scheduling.
```

Comment:
Is there anything fundamental preventing one from building spark-like graphs on top of dask graphs?